backend/routes/exchange-rates.js

```diff
@@ -0,0 +1,51 @@
+const fetch = require('node-fetch')
+const memoize = require('lodash/memoize')
+const { getLogger } = require('../utils/logger')
+const log = getLogger('routes.exchange-rates')
+
+const baseCurrency = 'USD'
+
+const currencies = [
+  // Tokens:
+  'ETH',
+  'DAI',
+  'OGN',
+  // Fiat:
+  'KRW',
+  'SGD',
+  'GBP',
+  'EUR',
+  'JPY',
+  'CNY'
+]
+
+/**
+ * Fetches and populates the exchange rates of all tokens
+ *
+ * @returns {{[token: String] => [rate: Number]}} An object of token => rate values
+ */
+async function fetchExchangeRatesRaw(target) {
+  target = currencies.indexOf(target) >= 0 ? '' : `,${target}`
+  const baseUrl = 'https://min-api.cryptocompare.com/data/price'
+  const allCurrencies = `${currencies.join(',')}${target}`
+  const exchangeURL = `${baseUrl}?fsym=${baseCurrency}&tsyms=${allCurrencies}`
+
+  const response = await fetch(exchangeURL)
+  if (!response.ok) {
+    throw new Error(response.error)
+  }
+  log.info('Fetched exchange rates OK')
+  return await response.json()
+}
+
+const fetchExchangeRates = memoize(fetchExchangeRatesRaw)
```

Comment:
Note (I found out today while looking at another bug on code using memoize): in case of an async function, memoize caches the Promise and therefore potentially the exception returned by the method. Let's say there is a network bleep or the crypto-compare API for some reason returned a temporary error, the memoize cache will return that error for 5min until the cache gets cleared. An alternative could be to have a wrapper function that clears the cache in case fetchExchangeRatesRaw returned an error.
Something like this:
```
const fetchExchangeRatesCache = memoize(fetchExchangeRatesRaw)

async function fetchExchangeRates(target) {
  try {
     fetchExchangeRatesCached(target)
  } catch(e) {
     fetchExchangeRatesCache.cache.delete(target)
     throw e
  }
```